handler/preprocess.sh

```diff
@@ -35,6 +36,48 @@ echo "finding dicom directories"
 ./find_dicomdir.py $root > $root/dcm2niix.list
 cat $root/dcm2niix.list
 
+# There can be overlap between the dicom directories and the pet directories
+# we want to use dcm2niix4pet for the pet directories, and dcm2niix for the dicom directories
+# but if and only if the conversion library PET2BIDS is installed
+PET2BIDS_INSTALLED=`which dcm2niix4pet > /dev/null && echo $?`
+
+if [ $PET2BIDS_INSTALLED -eq 0 ]; then
+
+    echo "PET2BIDS is installed, using dcm2niix4pet for PET directories found in root dir ${root}"
+    function rundcm2niix4pet {
+        #note.. this function runs inside $root (by --wd $root)
+
+        path=$1
+
+        echo "----------------------- $path ------------------------"
+        timeout 3600 dcm2niix4pet $path
+
+        #all good
+        echo $path >> pet2bids.done
+    } 
+
+    export -f rundcm2niix4pet
+
+    # now we do a little magic to run pet2bids if it's installed
+    ./find_petdir.py $root > $root/pet2bids.list
+    echo "Found PET directories:"
+    cat $root/pet2bids.list
+
+    # remove pet directories from dcm2niix list
+    echo "Removing PET directories from dcm2niix list"
+    grep -v -x -f <(sort pet2bids.list) <(sort dcm2niix.list) > dcm2niix.list
+    
+    # run pet2bids
+    true > $root/pet2bids.done
+
+    if [ $OSTYPE = "darwin" ]; then
```

Comment:
It may be good if we start phasing out these checks, so we focus on the Docker environment, wyt @dlevitas ?

Reply:
Agreed, these aren't necessary and can be phased out

Reply:
For production that makes sense, but I would strongly encourage you to keep them as someone who is forced to develop on Mac.

Reply:
I know, I should use development containers, but it's often better to be able to just run the script without any added fanfare.

Reply:
The thing is that you will end up having to run on the container, to test if the Linux version is still working.

What we can do is to ease the container dev process, for example like this: https://code.visualstudio.com/docs/devcontainers/containers
and/or create scripts that will smooth things out.

Reply:
Don't push that evil on me Anibal, I just installed parallel with brew.